app/src/main/java/org/wikiedufoundation/wikiedudashboard/ui/coursedetail/students/view/StudentListFragment.kt

```diff
@@ -42,37 +39,43 @@ class StudentListFragment : Fragment(), StudentListView {
         url = arguments?.getString("url", null).toString()
 
         studentListRecyclerAdapter = StudentListRecyclerAdapter(R.layout.item_rv_students) { openStudentProfile(it) }
+        initializeRecyclerView()
+        setData()
+        initializeProgressBar()
+        initializeToaster()
+    }
 
+    private fun initializeRecyclerView() {
         recyclerStudentList.apply {
             layoutManager = LinearLayoutManager(context)
             setHasFixedSize(true)
             adapter = studentListRecyclerAdapter
         }
-
-        url.let { studentListPresenter.requestStudentList(it) }
     }
 
-    override fun showProgressBar(show: Boolean) {
-        progressBar?.visibility = if (show) {
-            View.VISIBLE
-        } else {
-            View.GONE
-        }
+    private fun initializeProgressBar() {
+        studentsViewModel.progressbar.observe(this, androidx.lifecycle.Observer {
+            progressBar.visibility = if (it) View.VISIBLE else View.GONE
```

Comment:
You can use the ternary operator here in future. 

progressBar.visibility = it? View.VISIBLE : View.GONE

It looks more clean to me.

Reply:
Thanks will take note of it